jib-core/src/main/java/com/google/cloud/tools/jib/configuration/Port.java

```diff
@@ -32,40 +31,65 @@
     Protocol(String stringRepresentation) {
       this.stringRepresentation = stringRepresentation;
     }
+  }
 
-    @Override
-    public String toString() {
-      return stringRepresentation;
-    }
+  /**
+   * Create a new {@link Port} with TCP protocol.
+   *
+   * @param port the port number
+   * @return the new {@link Port}
+   */
+  public static Port tcp(int port) {
+    return new Port(port, Protocol.TCP);
+  }
 
-    /**
-     * Returns the {@link Protocol} given its string representation.
-     *
-     * @param protocolString the case insensitive string (e.g. "tcp", "udp")
-     * @return the {@link Protocol} with the corresponding name, or {@code Protocol#TCP} by default
-     *     if invalid
-     */
-    public static Protocol getFromString(@Nullable String protocolString) {
-      if (protocolString == null) {
-        return TCP;
-      }
-      return UDP.toString().equalsIgnoreCase(protocolString) ? UDP : TCP;
-    }
+  /**
+   * Create a new {@link Port} with UDP protocol.
+   *
+   * @param port the port number
+   * @return the new {@link Port}
+   */
+  public static Port udp(int port) {
+    return new Port(port, Protocol.UDP);
+  }
+
+  /**
+   * Gets a {@link Port} with protocol parsed from the string form {@code protocolString}. Unknown
+   * protocols will default to TCP.
+   *
+   * @param port the port number
+   * @param protocolString the case insensitive string (e.g. "tcp", "udp")
+   * @return the {@link Port}
+   */
+  public static Port parseProtocol(int port, String protocolString) {
+    Protocol protocol =
+        Protocol.UDP.toString().equalsIgnoreCase(protocolString) ? Protocol.UDP : Protocol.TCP;
+    return new Port(port, protocol);
   }
 
   private final int port;
-  private final Protocol protocol;
+  private final String protocol;
```

Comment:
If we're representing this as a string internally now I don't see how the enum is necessary anymore. Now `udp(int port)` and `tcp(int port)` might as well just `return new Port(port, "udp")` and `return new Port(port, "tcp")` respectively. It's especially weird in `parseProtocol` to be taking `protocolString`, converting it to an enum, passing that into the constructor, and immediately converting it back to a string.